### ISSUE TITLE
Cannot read property layer of undefined when passing a ref to Layer

### ISSUE THREAD

**Issue by Esya** [grommet/grommet#5313]
Passing a `ref` to Layer to wrap it using body-scroll-lock works with 2.17.2 but fails with 2.17.3 

### Expected Behavior

It should build and run fine 

### Actual Behavior

```
    TypeError: Cannot read property 'layer' of undefined 

      The above error occurred in the <ForwardRef> component: 
       
          at /home/esya/Git/oasis-wallet/node_modules/grommet/components/Layer/LayerContainer.js:39:25 
          at Layer (/home/esya/Git/oasis-wallet/node_modules/grommet/components/Layer/Layer.js:27:23) 
          at ResponsiveLayer (/home/esya/Git/oasis-wallet/src/app/components/ResponsiveLayer/index.tsx:30:44) 
```

### URL, screen shot, or Codepen exhibiting the issue

```typescript
import { disableBodyScroll, enableBodyScroll } from 'body-scroll-lock'
import { Layer, LayerExtendedProps } from 'grommet'
import { useCallback, useRef } from 'react'

function useRefWithCallback(onMount, onUnmount) {
  const nodeRef = useRef(null)

  const setRef = useCallback(
    node => {
      if (nodeRef.current) {
        onUnmount(nodeRef.current)
      }
      nodeRef.current = node
      if (nodeRef.current) {
        onMount(nodeRef.current)
      }
    },
    [onMount, onUnmount],
  )

  return setRef
}

/**
 * Wraps grommet's layer with body-scroll-lock
 * @param props Grommet Layer props
 */
export function ResponsiveLayer(props: LayerExtendedProps) {
  const layerRef = useRefWithCallback(
    node => disableBodyScroll(node),
    node => enableBodyScroll(node),
  )

  return <Layer {...props} ref={layerRef} />
}
```

### Your Environment

- Grommet version: 2.17.3

**Comment by li5tun** [comment]
yeah, after upgrade v2.17.3, I have the same problem running test.

**Comment by jlaluces123** [comment]
Hello @Esya, I tried to replicate the error you're receiving in a codesandbox [here](https://codesandbox.io/s/blissful-merkle-3lpbi?file=/src/App.js). I couldn't manage to replicate the issue on my end and was hoping if you could provide a bit more context on how / when the issue comes up.

**Comment by jlaluces123** [comment]
Hey @Esya, wanted to check in after a little bit of time. Would it still be possible to get more context on the issue and how it comes up?

**Comment by Esya** [comment]
Hi @jlaluces123 - the snippet I provided should work - your sandbox is plain vanilla javascript, this is a typescript error introduced by Grommet 2.17.3

**Comment by Esya** [comment]
Actually, I'm facing the isssue with even just this simple jest test : 

```typescript
import { render } from '@testing-library/react'
import { Layer } from 'grommet'
import * as React from 'react'

const renderComponent = () => render(<Layer />)

describe('<Layer />', () => {
  it.only('should match snapshot', () => {
    const component = renderComponent()
    expect(component.container.firstChild).toMatchSnapshot()
  })
})
```

It turns out that Layer could work outside of <Grommet>'s provider but now requires a context to run, so it is fixed by doing it as such : 

```typescript
const renderComponent = () => render(<Grommet><Layer /></Grommet>)
```

**Comment by MikeKingdom** [comment]
I believe this one is fixed by #5524
